src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Segment.java

```diff
@@ -0,0 +1,58 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+public abstract class Segment {
+    protected static final String PLUS_SEPARATOR = "+";
+    protected final static String COLON_SEPARATOR = ":";
+    private static final String TERMINATOR = "'";
+
+    protected static <T> List<T> removeEmptyTrailingFields(List<T> list, Predicate<T> predicate) {
```

Comment:
This method name is only appropriate if the `predicate` given to it meets the semantics of "empty" (or _not_-empty, given the implementation).

Perhaps
```suggestion
    protected static <T> List<T> removeTrailingFields(List<T> list, Predicate<T> isNonTrailing) {
```

Reply:
I'm reluctant to change method names in case we need to extract other modules in to the project.

Reply:
Yep, fair enough!

If you agree with the suggestion, we can still make a note to change it when we're satisfied we're finish copying modules over. Add a comment like `// TODO rename as per https://github.com/nhsconnect/integration-adaptor-lab-results/pull/6#discussion_r555038977`

But you're still free to disagree entirely, of course! In that case we'd ask for another's opinion.

Reply:
I've pushed that change up - I agree it should be renamed